code/game/objects/structures.dm

```diff
@@ -267,9 +263,9 @@
 	if(!dam)
 		return
 
-	damage = max(0, damage + dam)
+	obj_integrity = max(0, obj_integrity - dam)
```

Comment:
If you want to preserve behavior do:
`obj_integrity = min(max_integrity, obj_integrity - dam)`
Else just go with the unsanitized
`obj_integrity -= dam)`

Reply:
it'll get replaced with take_damage